app/modules/cart/repository.py

```diff
@@ -1,22 +1,95 @@
+from sqlalchemy import select
+from sqlalchemy.exc import SQLAlchemyError
 from sqlalchemy.ext.asyncio import AsyncSession
 
+from app.common.exceptions.app_exceptions import DatabaseOperationException, EntityNotFoundException
+
 from .models import CartItem
 from .repository_interface import CartItemRepositoryInterface
 
 
 class CartItemRepository(CartItemRepositoryInterface):
+    model_class = CartItem
+
     def __init__(self, session: AsyncSession):
         self.session = session
 
     async def insert(self, cart_item: CartItem) -> CartItem:
-        self.session.add(cart_item)
-        await self.session.commit()
-        await self.session.refresh(cart_item)
-        return cart_item
+        try:
+            self.session.add(cart_item)
+            await self.session.commit()
+            await self.session.refresh(cart_item)
+            return cart_item
+        except SQLAlchemyError as e:
+            await self.session.rollback()
+            raise DatabaseOperationException(
+                operation="create",
+                message=str(e),
+                data={"user_id": cart_item.user_id, "product_id": cart_item.product_id},
+            )
 
     async def bulk_insert(self, cart_items: list[CartItem]) -> list[CartItem]:
-        self.session.add_all(cart_items)
-        await self.session.commit()
-        for cart_item in cart_items:
-            await self.session.refresh(cart_item)
-        return cart_items
+        try:
+            self.session.add_all(cart_items)
+            await self.session.commit()
+            for cart_item in cart_items:
+                await self.session.refresh(cart_item)
+            return cart_items
+        except SQLAlchemyError as e:
+            await self.session.rollback()
+            raise DatabaseOperationException(
+                operation="create",
+                message=str(e),
+                data={"items": [{"user_id": cart_item.user_id, "product_id": cart_item.product_id} for cart_item in cart_items]},
+            )
+
+    async def find_all(self, user_id: int) -> list[CartItem]:
+        try:
+            stmt = select(self.model_class).where(self.model_class.user_id == user_id)
+            result = await self.session.execute(stmt)
+            return result.scalars().all()
+        except SQLAlchemyError as e:
```

Comment:
we usually handle DB exceptions at a higher layer (e.g., service layer/usecase) rather than the repository itself, to keep our repositories focused on data access logic and avoid coupling them with application-specific error handling.